mock-certify-plugin/src/main/java/io.mosip.certify.mock.integration/service/MockVCIssuancePlugin.java

```diff
@@ -235,10 +239,47 @@ private static String getUTCDateTime() {
 	@Override
 	public VCResult<String> getVerifiableCredential(VCRequestDto vcRequestDto, String holderId,
 													Map<String, Object> identityDetails) throws VCIExchangeException {
+		String accessTokenHash = identityDetails.get(ACCESS_TOKEN_HASH).toString();
+        String documentNumber;
+        try {
+			documentNumber = getIndividualId(getUserInfoTransaction(accessTokenHash));
+		} catch (Exception e) {
+			log.error("Error getting documentNumber", e);
+			throw new VCIExchangeException(ErrorConstants.VCI_EXCHANGE_FAILED);
+		}
+
+		if(vcRequestDto.getFormat().equals("mso_mdoc")){
+			VCResult<String> vcResult = new VCResult<>();
+			String mdocVc = null;
+			try {
+				 mdocVc = new io.mosip.certify.mock.integration.mocks.MdocGenerator().generate(mockDataForMsoMdoc(documentNumber),holderId, issuerKeyAndCertificate);
+			} catch (Exception e) {
+                log.error("Exception on mdoc creation", e);
+				throw new VCIExchangeException(ErrorConstants.VCI_EXCHANGE_FAILED);
+			}
+			vcResult.setCredential(mdocVc);
+			vcResult.setFormat("mso_mdoc");
+			return  vcResult;
+		}
+        log.error("not implemented the format {}", vcRequestDto);
 		throw new VCIExchangeException(ErrorConstants.NOT_IMPLEMENTED);
 	}
 
+	private Map<String, Object> mockDataForMsoMdoc(String documentNumber) {
+		Map<String, Object> data = new HashMap<>();
+		log.info("Setting up the data for mDoc");
+		data.put("family_name","Agatha");
+		data.put("given_name","Joseph");
+		data.put("birth_date", "1994-11-06");
+		data.put("issuing_country", "IN");
+		data.put("document_number", documentNumber);
+		data.put("driving_privileges",new HashMap<>(){{
+			put("vehicle_category_code","A");
+		}});
```

Comment:
There is a consistency, the LD VC generator from this class will create a VC with data from mock identity system, but this has hardcoded code. I accept this is a mock, but if we create a new vciplugin class with mdoc impl, we might be little better.

Reply:
hello @vishwa-vyom 
Are we suggesting like `MdocVCIPlugin` class which will encapsulate both the data and generation logic?

Reply:
Lets create a new Plugin as MDocMockVCIPlugin.java and move all the code to this class

Reply:
Is MDocMockVCIPlugin also expected to implement VCIssuancePlugin  and when mdoc is required config will be updated to point to the MdocVCIPlugin?
If this change is coming we will not be able to issue both ldp_vc as well as mdoc from same plugin